src/app/modules/auth/components/reset-password/reset-password.component.ts

```diff
@@ -0,0 +1,93 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { ActivatedRoute } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { CustomErrorStateMatcher } from 'src/app/modules/core/utilities/error-state-matcher';
+import { SignInSignUpService } from '../../sign-in-sign-up.service';
+
+@Component({
+  selector: 'app-reset-password',
+  templateUrl: 'reset-password.component.html',
+  styleUrls: ['reset-password.component.css'],
+})
+export class ResetPasswordComponent implements OnInit {
+  public resetPasswordForm: FormGroup;
+  public matcher: CustomErrorStateMatcher;
+  private routeSubscription: Subscription;
+  private compositeField: string;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private activatedRoute: ActivatedRoute,
+    private snackBar: MatSnackBar,
+    private signInSignUpService: SignInSignUpService
+  ) {}
+
+  ngOnInit() {
+    this.createRouteSubscription();
+    this.initializeResetPasswordForm();
+    this.matcher = new CustomErrorStateMatcher();
+  }
+
+  private createRouteSubscription() {
+    this.routeSubscription = this.activatedRoute.params.subscribe((params) => {
```

Comment:
what are you getting as uid? user Id or email / phone number?

Reply:
If a user will be redirected from mail link to the application his/her uid will be passed as URL param